lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -55,7 +55,7 @@ def initialize(cmd, debug, pipe_timeout = 30)
         # wait for the pipe server to signal ready, and fail if no response in 10 seconds
 
         # wait up to 30 seconds in 0.2 second intervals to be able to open the pipe
-        sleep_interval = 0.2
+        sleep_interval = 0.01
         (pipe_timeout / sleep_interval).to_int.times do
```

Comment:
This match ```(pipe_timeout / sleep_interval).to_int``` comes out to zero if ```pipe_timeout``` is ever allowed to get to this point and has a value less than ```sleep_interval```. Is this something we feel we can ignore, or do want to test and throw a descriptive error if that occurs? This risk here is that since external users won't usually plumb the code deep enough to know what a safe value is, they could potentially pass anything in here as a value.

Reply:
Puppet code should be the only caller here, so I don't *think* we'll need to be overly defensive.

Reply:
Also, I think calling `0.times` will just cause an auto failure... so might be a sort-of fix for the sleep duration issue below?

Reply:
Sorry, by "auto failure" I mean it will never assign anything to `@pipe`, so we should always exit as if we timed out, right?

Reply:
Just verified this: if we set the timeout to something _lower_ than the `sleep_duration, it is always treated as if there is a timeout waiting for the pipe.